src/libraries/System.Text.Json/gen/JsonSourceGenerator.3.9.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
Nit: filename seems a bit odd to me. How about something like `JsonSouceGenerator.Roslyn39.cs`?

Reply:
or perhaps `RoslynV3` and `RoslynV4`.